pkg/asset/operators/kubeCoreOperator.go

```diff
@@ -0,0 +1,127 @@
+
+// Dependencies returns all of the dependencies directly needed by an
+// kubeCoreOperator asset.
+func (kco *kubeCoreOperator) Dependencies() []asset.Asset {
+	return []asset.Asset{
+		kco.installConfigAsset,
+	}
+}
+
+// Generate generates the kube-core-operator-config.yml files
+func (kco *kubeCoreOperator) Generate(dependencies map[asset.Asset]*asset.State) (*asset.State, error) {
+	ic, err := installconfig.GetInstallConfig(kco.installConfigAsset, dependencies)
+	if err != nil {
+		return nil, err
+	}
+	kco.installConfig = ic
+
+	// installconfig is ready, we can create the core config from it now
+	coreConfig, err := kco.coreConfig()
+	if err != nil {
+		return nil, err
+	}
+
+	data, err := yaml.Marshal(coreConfig)
+	if err != nil {
+		return nil, fmt.Errorf("Failed to marshal core config: %v", err)
+	}
+	state := &asset.State{
+		Contents: []asset.Content{
+			{
+				Name: filepath.Join(kco.directory, "kube-core-operator-config.yml"),
+				Data: data,
+			},
+		},
+	}
+	return state, nil
+}
+
+func (kco *kubeCoreOperator) coreConfig() (*kubecore.OperatorConfig, error) {
+	coreConfig := kubecore.OperatorConfig{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: kubecore.APIVersion,
+			Kind:       kubecore.Kind,
+		},
+	}
+	coreConfig.ClusterConfig.APIServerURL = kco.getAPIServerURL()
+	coreConfig.AuthConfig.OIDCClientID = authConfigOIDCClientID
+	coreConfig.AuthConfig.OIDCIssuerURL = kco.getOicdIssuerURL()
+	coreConfig.AuthConfig.OIDCGroupsClaim = authConfigOIDCGroupsClaim
+	coreConfig.AuthConfig.OIDCUsernameClaim = authConfigOIDCUsernameClaim
+
+	ip, err := cidr.Host(&kco.installConfig.Networking.ServiceCIDR, 10)
+	if err != nil {
+		return nil, err
+	}
+	coreConfig.DNSConfig.ClusterIP = ip.String()
+
+	coreConfig.CloudProviderConfig.CloudConfigPath = ""
+	coreConfig.CloudProviderConfig.CloudProviderProfile = k8sCloudProvider(kco.installConfig.Platform)
+
+	coreConfig.RoutingConfig.Subdomain = kco.getBaseAddress()
+
+	coreConfig.NetworkConfig.ClusterCIDR = kco.installConfig.Networking.PodCIDR.String()
+	coreConfig.NetworkConfig.ServiceCIDR = kco.installConfig.Networking.ServiceCIDR.String()
+	coreConfig.NetworkConfig.AdvertiseAddress = networkConfigAdvertiseAddress
+	coreConfig.NetworkConfig.EtcdServers = kco.getEtcdServersURLs()
+
+	return &coreConfig, nil
+}
+
+func (kco *kubeCoreOperator) getAPIServerURL() string {
+	return fmt.Sprintf("https://%s-api.%s:6443", kco.installConfig.ClusterName, kco.installConfig.BaseDomain)
+}
+
+func (kco *kubeCoreOperator) getEtcdServersURLs() string {
+	return fmt.Sprintf("https://%s-etcd.%s:2379", kco.installConfig.ClusterName, kco.installConfig.BaseDomain)
+}
+
+func (kco *kubeCoreOperator) getOicdIssuerURL() string {
+	return fmt.Sprintf("https://%s.%s/identity", kco.installConfig.ClusterName, kco.installConfig.BaseDomain)
+}
+
+func (kco *kubeCoreOperator) getBaseAddress() string {
+	return fmt.Sprintf("%s.%s", kco.installConfig.ClusterName, kco.installConfig.BaseDomain)
+}
+
```

Comment:
all these functions seem like be used by multiple assets. there should be a common source of finding these.